File: RanyAzevedo/week_01/Homework -MTA/js/MTA.js
//planTrip('N', 'Times Square', '6', '33rd');
// This is only a suggested function name and signature.

// console.log() shows output similar to this:
// "Your must travel through the following stops on the N line: 34th, 28th,
// 23rd, Union Square."
// "Change at Union Square."
// "Your journey continues through the following stops: 23rd, 28th, 33rd."
// "7 stops in total."

//slice  reverse  indexOf

var lines = {
lineN: ['34th', '28thN', '23rdN', 'Union Square','8thN'],
lineL: ['8thL','6th','Union Square', '3rd','1st'],
line6: ['Grand Central', '33rd', '28th6', '23rd6', 'Union Square', 'Astor Place']
}


var goStops = [];
var backStops = [];

//Trip in one Line

//1

var goForwards = function(firstLine, initialStop, finalStop){
  goStops = [];
  var goForwards = lines[firstLine];
  var initialIndex = goForwards.indexOf(initialStop);
  var finalIndex = goForwards.indexOf(finalStop);
  for (var i = initialIndex; i <= finalIndex; i++) {
    goStops.push(goForwards[i]);
  }
  var message = "You must travel trough the following stops on the " ;
  console.log(message + firstLine + ": " + goStops);
  return goStops;
};

//2

var goBackwards = function (firstLine, initialStop, finalStop) {
  backStops = [];
  var goBackwards = lines[firstLine];
  var initialIndex = goBackwards.indexOf(initialStop);
  var finalIndex = goBackwards.indexOf(finalStop);
  for (var i = initialIndex; i >= finalIndex ; i--) {
      backStops.push(goBackwards[i]);
  }
  var message = "You must travel trough the following stops on the  " ;
  console.log(message + firstLine + ": " + backStops);
  return backStops;
};

//  Two Lines

var planTrip = function(firstLine, initialStop, secondLine, finalStop) {
  goStops = [];
  backStops = [];
    var planTrip = lines[firstLine];
    var initialIndex = planTrip.indexOf(initialStop);
    var finalIndex = planTrip.indexOf(finalStop);
    var unionIdenx1 = planTrip.indexOf('Union Square');

//Same Lines

  if ( firstLine === secondLine ){
    if ( initialIndex < finalIndex) {
     goForwards(firstLine, initialStop, finalStop);
    } else {
     goBackwards(firstLine, initialStop, finalStop);
  }
  var totalAmountOfStops = goStops.length + backStops.length;
  console.log("Total amount of stops: ", totalAmountOfStops);
  return "Trip Finished";

 }
// Diferents Lines

// 1st - go from initialStop to 'Union Square';

  if (initialIndex < unionIdenx1) {
    goForwards(firstLine, initialStop, "Union Square");
  } else {
    goBackwards(firstLine, initialStop, "Union Square");
  }
  console.log('Change at Union Square.');


// 2nd -  go from 'Union Square' to finalStop;

  if (finalIndex < unionIdenx1) {
    goForwards(secondLine, "Union Square", finalStop);
  } else {
    goBackwards(secondLine, "Union Square", finalStop);
  }
  var totalAmountOfStops = goStops.length + backStops.length;
  console.log("Total amount of stops: ", totalAmountOfStops);
  return "Trip Finished";
};

File: RanyAzevedo/week_04/MTA-Ruby/mta.rb

@stops1 = [];
@stops2 = [];

@lines = {
"n" => ['34th', '28thN', '23rdN', 'Union Square','8thN'],
"l" => ['8thL','6th','Union Square', '3rd','1st'],
"6" => ['Grand Central', '33rd', '28th6', '23rd6', 'Union Square', 'Astor Place']
}

print "Enter the line you are in (n, l , 6)? "
@first_line = gets.chomp
print "Enter the station you are: "
@initial_station = gets.chomp
print "Enter the line you want to go (n,l,6): "
@second_line = gets.chomp
print "Enter the station you want to go: "
@final_station = gets.chomp

# FROM initial_index TO  UNION SQUARE

def between_lines line, second_line
  initial_index = line.index(@initial_station)
  final_index = second_line.index(@final_station)
  union_first = line.index("Union Square")
  union_second = second_line.index("Union Square")

  #SAME LINE
  if @first_line == @second_line
    line[initial_index..final_index].each {|stops| @stops1 << stops }
    line[final_index..initial_index].reverse_each {|stops| @stops1 << stops }
    puts @stops1
    puts "Total of #{@stops1.length - 2} stops"
    puts "Trip Finished!"
    return
  end

  # FIRST LINE - FROM initial_index TO  UNION SQUARE
  if initial_index < union_first
    line[initial_index..union_first].each {|stops| @stops1 << stops}
  else
    line[union_first..initial_index].reverse_each {|stops| @stops1 << stops }
  end
    puts @stops1
    puts  "Change at Union Square"

  # SECOND LINE - FROM UNION SQUARE TO final_index
  if final_index < union_second
    second_line[final_index..union_second].reverse_each {|stops| @stops2 << stops }
  else
    second_line[union_second..final_index].each {|stops| @stops2 << stops}
  end
    puts @stops2
    puts " Total of #{@stops1.length + @stops2.length - 2} stops"
    puts "Trip Finished!"
end

between_lines @lines[@first_line], @lines[@second_line]
